core/src/main/scala/org/apache/spark/status/api/v1/api.scala

```diff
@@ -527,13 +527,53 @@ case class StackTrace(elems: Seq[String]) {
 }
 
 case class ThreadStackTrace(
-    val threadId: Long,
-    val threadName: String,
-    val threadState: Thread.State,
-    val stackTrace: StackTrace,
-    val blockedByThreadId: Option[Long],
-    val blockedByLock: String,
-    val holdingLocks: Seq[String])
+    threadId: Long,
+    threadName: String,
+    threadState: Thread.State,
+    stackTrace: StackTrace,
+    blockedByThreadId: Option[Long],
+    blockedByLock: String,
+    @deprecated("using synchronizers and monitors instead", "4.0.0")
+    holdingLocks: Seq[String],
+    synchronizers: Seq[String],
+    monitors: Seq[String],
+    lockName: Option[String],
+    lockOwnerName: Option[String],
+    suspended: Boolean,
+    inNative: Boolean) {
+
+  /**
+   * Returns a string representation of this thread stack trace
+   * w.r.t java.lang.management.ThreadInfo(JDK 8)'s toString.
+   *
+   * TODO(Kent Yao): Considering 'daemon', 'priority' from higher JDKs
+   *
+   * TODO(Kent Yao): Also considering adding information os_prio, cpu, elapsed, tid, nid, etc.,
+   *   from the jstack tool
+   */
+  override def toString: String = {
+    val sb = new StringBuilder()
+    val basic = "\"" + threadName + "\" Id=" + threadId + " "  + threadState
```

Comment:
```
val basic = s""""$threadName" Id=$threadId $threadState"""
sb.append(basic)
```
or

```
sb.append("\"").append(threadName).append("\" Id=").append(threadId).append(" ").append(threadState)
// sb.append("\"", threadName, "\" Id=", threadId, " ", threadState) multiple parameters api needs to wait until only Scala 2.13+ is supported.
```
The former has better readability, while the latter can avoid creating the intermediate string `basic`.

Reply:
I choose another simple way `val sb = new StringBuilder(s""""$threadName" Id=$threadId $threadState""")`